### DISCUSSION TITLE
Modern search webpart ignoring data source ID for certain GEO users hence showing despcrepancies in results

### DISCUSSION THREAD

**Discussion by ABHISHBH** [microsoft-search/pnp-modern-search#3538] [Q&A]
Hello, i have been using PnP Modern Search web part v4.7.0 to get files from a certain document library in sharepoint. It is although working fine for INDIA users but however certain users from other GEO locations(USA) have been getting incorrect results.
When we checked in their dev tool network tab, it became clear to us that the result source ID is getting changed for those users thus rendering incorrect data from local sharepoint site, but we are targetting a particular content type ID in the result source.
Users are accessing the same site collection URL across all GEO locations.

**Comment by wobba** [comment]
Without knowing the exact setup of result sources, usage of them and any query rules set up it is hard to figure out what happens. If all users search from the same site URL, then they should get the same results. And you don't call out what the incorrect results are.

**Reply by ABHISHBH** [reply]
result source ID is getting disregared, hence undesired results are getting displayed from all over sharepoint. so yes, those are to me incorrect results and not the data sets i need with regard to the result source.

**Reply by ABHISHBH** [reply]
Please suggest a workaround if anyone has faced this issue.

**Reply by kasperbolarsen** [reply]
You are referring to a Search Result source you have created yourself? Will it be possible to test if the issue will be resolved by using the Local SharePoint Results (8413cd39-2156-4e00-b54d-11efd9abdb89) ?

**Reply by wobba** [reply]
@ABHISHBH you need to share your exact setup, and possibly the request and response JSON for us to properly guide you on what goes wrong.

**Reply by ABHISHBH** [reply]
We have created our own result source and configured it for pnp search result webpart. It is working for all users except US users. For US users, when we checked in network tab, post query, resultsource id is automatically getting changed to local sharepoint. Also when we created classic page with classic search result webpart provided by MS, this works perfectly and brings the result from custom result source.

**Reply by ABHISHBH** [reply]
Hi Kasperbolarsen,
i didnt get the question actually but we donot need to target data from local sharepoint. we just need to target a specified custom result source. and it has been working fine. just recently only for few users from other GEO locations the result source is getting disregarded and data is getting fetched from all sharepoint sites. whereas for other users it is targetting the required result source and getting the proper data.

**Reply by kasperbolarsen** [reply]
Hi @ABHISHBH , the custom result source is build upon the localSharePointResult , so I was just wondering if the issue would be resolved by using localSharePointResults and modifying the query accordingly?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/c5bb122f-96d7-4324-b9a8-52c2f43ee3ef)
